controllers/openstackmachine_controller.go

```diff
@@ -353,7 +354,32 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, logger
 			handleUpdateMachineError(logger, openStackMachine, errors.Errorf("Floating IP cannot be got or created: %v", err))
 			return ctrl.Result{}, nil
 		}
-		err = computeService.AssociateFloatingIP(instance.ID, fp.FloatingIP)
+
+		// There is some possibility that floating IP is created but not able to be find right after that
+		// so give a retry here, see #927 for further information.
+		backoff := wait.Backoff{
+			Steps:    10,
+			Duration: time.Second,
+			Factor:   1.25,
+			Jitter:   0.1,
+		}
+		err = wait.ExponentialBackoff(backoff, func() (bool, error) {
+			if err := computeService.AssociateFloatingIP(instance.ID, fp.FloatingIP); err != nil {
+				if apierrors.IsNotFound(err) {
```

Comment:
I think you wanted to use `capoerrors "sigs.k8s.io/cluster-api-provider-openstack/pkg/utils/errors"` 's `IsNotFound?

Also maybe it's worth to `return false, nil` on `capoerrors.IsRetryable`? :-)

Reply:
I am not sure IsRetryable is suitable here 
for example, conflict or badrequest we should not retry anymore